tianshou/highlevel/module/actor.py

```diff
@@ -159,6 +179,9 @@ def create_module(self, envs: Environments, device: TDevice) -> BaseActor:
             device=device,
         ).to(device)
 
+    def create_dist_fn(self, envs: Environments) -> TDistFnDiscrOrCont | None:
```

Comment:
Return type should be `None`

Reply:
I don't necessarily agree with that.
While Python can use covariant return types, this doesn't mean that they have to be used or that their use is necessarily advantageous.

In fact, I would argue that the default should be for the interface to remain unchanged in subclasses (unless there is a specific reason to change it). As usual, my decision was deliberate and not accidental.
If the more specific return type is used, then, notably, a user cannot inherit from the class and return anything other than the more specific type (i.e. None) without violating the Liskov substitution principle.
The only reason to use a more specific return type is a case where users implement against specialisations rather than the abstraction and thus might care about having more specific return type information. This isn't the case here; all implementations use the abstraction.

Reply:
Ah, I see, ok. In one of the factories you did use a stricter type, but I guess that was also deliberate. Feel free to merge

Reply:
Well, my decision in this case was only *somewhat* deliberate, because a key part is that the unchanged signature had already been auto-generated, and, as explained above, I figured that nobody would care about having a more specific return type in this case, because only the abstraction really matters.